src/CoreBundle/MaintenanceMode/MaintenanceModeService.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\MaintenanceMode;
+
+use ChameleonSystem\CoreBundle\Exception\MaintenanceModeErrorException;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use esono\pkgCmsCache\CacheInterface;
+
+class MaintenanceModeService implements MaintenanceModeServiceInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    public function __construct(Connection $connection, CacheInterface $cache)
+    {
+        $this->connection = $connection;
+        $this->cache = $cache;
+    }
+
+    public function isActivated(): bool
+    {
+        return true === file_exists(PATH_MAINTENANCE_MODE_MARKER);
+    }
+
+    public function isActivatedInDb(): bool
+    {
+        return \TdbCmsConfig::GetInstance()->fieldShutdownWebsites;
+    }
+
+    public function activate(): void
+    {
+        try {
+            $this->connection->executeUpdate("UPDATE `cms_config` SET `shutdown_websites` = '1'");
+
+            $this->cache->callTrigger('cms_config');
+        } catch (DBALException $exception) {
+            throw new MaintenanceModeErrorException('Cannot save maintenance mode flag in database', 0, $exception);
```

Comment:
I can't answer to outdated conversations, so I start anew: Anything looking like a sentence should end with a dot.